File: 4C   |/Solution1_4C.cpp
//This solution will Give me TLA if the TLA limit is half

#include<iostream>
#include<list>
#include<utility>
using namespace std;
int main(){
    int times;
    string str;
    list<pair<string,int>> lt;
    cin>>times;
    cin.ignore();
    while(times--){
        getline(cin,str);
        bool breakk=false;
        for(auto& i: lt){
            if(i.first==str){
                ++(i.second);
                cout<<i.first<<i.second<<"\n";
                breakk=true;
                break;
            }
        }if(breakk==false){
            lt.push_back({str,0});
            cout<<"OK\n";
        }  
    }
    return 0;
}

File: 4C   |/Solution1_4C.py
#This solution will Give me TLA if the TLA limit is half
times=int(input())
lt=[]
ans=[]
for _ in range(times):
    l=input()
    lt.append(l)
for item in range(times):
    count=0
    i=item-1
    while(i>=0):
        
        if lt[item]==lt[i]:
            count+=1
        i-=1    
    if(count>0):
        ans.append(lt[item]+str(count))
    else:
        ans.append("OK")     

for _ in (ans):
    print(_)       
        
